### PR TITLE
Update the endpoint to run the eligibility rules for the Actions

### PR DESCRIPTION
**Pull request by Matthew-Collins-Defra**
Update the GET /parcels/{parcelRef}/actions endpoint to run the eligibility rules for the Actions

https://eaflood.atlassian.net/browse/FCEP-75

### CODE REVIEW

File: server/services/parcelActionsService.js [hunk 1]
```diff
@@ -1,7 +1,24 @@
-const actions = require('../../data/actions.json')
+const actionsService = require('./actionsService')
+const parcelService = require('./parcelService')
+const rulesEngine = require('../rules-engine')
 
 module.exports = {
   get: async function (parcelRef) {
-    return actions
+    const actions = actionsService.get()
```
Comment (markysoft) [inline server/services/parcelActionsService.js]: The actionService has an `async` method signature, but an `await` is not being used when the function is called.

Causing an error when calling the endpoint, i.e. a get on `http://localhost:3002/parcels/SD74445738/actions`

**Review by markysoft — CHANGES_REQUESTED** [review]
Missing await causing an error when calling the end point.

**Note**: there are several places in the code where methods are marked as async when they are not and vice versa, which can lead to unexpected consequences.

File: test/services/parcelActionsService.test.js [hunk 2]
```diff
@@ -1,17 +1,94 @@
...
+  return {
+    getByRef: jest.fn((parcelRef) => {
+      return mockParcelsToReturn.find(x => x.ref === parcelRef)
+    })
+  }
+})
+
+jest.mock('../../server/services/actionsService', () => {
+  return {
+    get: jest.fn(() => {
```
Comment (markysoft) [inline test/services/parcelActionsService.test.js]: the action service get function is marked as async, but the mock is returning an array

File: server/services/parcelActionsService.js (hunk 1 above)
Comment (Matthew-Collins-Defra) [reply server/services/parcelActionsService.js]: Thanks for that. Can't believe I missed that. Fixed now.

File: test/services/parcelActionsService.test.js (hunk 2 above)
Comment (Matthew-Collins-Defra) [reply test/services/parcelActionsService.test.js]: Fixed now, plus a further bug picked up once the tests were running correctly.

File: server/services/parcelActionsService.js [hunk 3]
```diff
@@ -1,7 +1,25 @@
...
+    const parcelData = parcelService.getByRef(parcelRef)
+    for (const action of actions) {
+      rulesEngine.resetEngine()
+      let actionPassed = false
+      const successFunc = function () {
+        actionPassed = true
+      }
+      await rulesEngine.doEligibilityRun(action.rules, [parcelData], { }, successFunc)
+      if (actionPassed) {
+        const returnAction = Object.assign({}, action)
```
Comment (markysoft) [inline server/services/parcelActionsService.js]: this could be done in one line with:
` const { rules, ...returnAction } = action`

**Review by markysoft — DISMISSED** [review]
Looks good, added a small comment on a code improvement, but happy for it to be merged as is.

**Review by markysoft — APPROVED** [review]
LGTM

**Pull request merged**